server/paymentProviders/stripe/webhook.ts

```diff
@@ -295,9 +451,96 @@ export const paymentIntentProcessing = async (event: Stripe.Event) => {
 
     sendEmailNotifications(order);
   });
-};
+}
+
+async function handleExpensePaymentIntentProcessing(event: Stripe.Event) {
+  const paymentIntent = event.data.object as Stripe.PaymentIntent;
+
+  const stripeAccount = event.account ?? config.stripe.accountId;
+
+  await sequelize.transaction(async transaction => {
+    const expense = await models.Expense.findOne({
+      where: {
+        data: { paymentIntent: { id: paymentIntent.id } }, // TODO(henrique): add index
+      },
+      transaction,
+      lock: transaction.LOCK.UPDATE,
+      include: [
+        { association: 'collective', required: true },
+        { association: 'fromCollective', required: true },
+      ],
+    });
+
+    if (!expense) {
+      logger.debug(`Stripe Webhook: Could not find Expense for Payment Intent ${paymentIntent.id}`);
```

Comment:
Logs get lost, I'd prefer to have a true warning if that happens

```suggestion
      reportMessageToSentry(`Stripe Webhook: Could not find Expense for Payment Intent ${paymentIntent.id}`, { extra: { event } });
```